code/modules/reagents/machinery/dispenser/reagent_tank.dm

```diff
@@ -57,6 +75,25 @@
 	if (N)
 		amount_per_transfer_from_this = N
 
+/obj/structure/reagent_dispensers/Move()
+	if (..() && modded)
+		leak(amount_per_transfer_from_this / 5 / reagents.get_viscosity())
+
+/obj/structure/reagent_dispensers/proc/leak(amount)
+	if (reagents.total_volume <= 0)
+		return
+
+	amount = min(amount, reagents.total_volume)
+
+	var/turf/T = get_turf(src)
+
+	var/obj/effect/decal/cleanable/chempuddle/CP = locate() in T
+	if(!CP)
+		CP = new(T)
+	reagents.trans_to_holder(CP.reagents, amount)
+	spawn(1)
+		CP.Spread()
```

Comment:
Should spreading as a result of adding reagents to a puddle, be handled by that puddle? (I.e., overriding `trans_to_holder()`)